### PR TITLE
Add build tests

### PR DESCRIPTION
**Pull request by obelisk**
Splitting this out from the large PR.

Adds tests to ensure the code builds correctly on macOS and Ubuntu operating systems. This should catch any breakage on macOS and Raspberry Pi OS/Linux systems.

### CODE REVIEW

**Comment by gebogebogebo** [comment]
I have a request.
Change to PR to merge into the develop branch.

**Comment by obelisk** [comment]
This just adds build tests, so PRs are automatically built on macOS and Ubuntu to ensure they work, this helps prevent the FidoKeyPi issue from a couple weeks ago.

We could merge it to the a develop branch but I'm curious as to why? This doesn't change any code, it just makes PRs autotest.

**Comment by obelisk** [comment]
Changed to develop as requested :) (for all my PRs)

**Review by gebogebogebo — COMMENTED** [review]
Written comment.

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  CARGO_TERM_COLOR: always
+
+jobs:
+  macos-build:
+    runs-on: macos-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Build
+      run: cargo build --examples --verbose
```
[the remark this answers is not in the document]
Comment (obelisk) [reply .github/workflows/rust.yml]: My mistake, I left in the example flag when this project has no examples. I have another branch where I'm working on porting all binaries in the examples folder to true examples, it's just not in this this diff.

**Comment by obelisk** [comment]
We can re-add the examples flag once they are all converted, probably after the refactor PR.

**Comment by obelisk** [comment]
@gebogebogebo You can see the tests run here on my own PR: https://github.com/obelisk/ctap-hid-fido2/pull/4

**Comment by gebogebogebo** [comment]
Great !!

**Pull request merged**